### PR TITLE
PIM-8950: PoC on Content Security Policy

### PR DESCRIPTION
**Pull request by jmleroux**
This PR add a Content Security Policy to prevent the execution of javascript `<script></script>` tags in the PIM, mainly to void direct execution in WYSIWYG editors. Sadly, we still have some legacy and legit scripts tags that still must be executed. For these tags, CSP allows to identify them as legit by adding them a nonce string that must correspond to the same nonce in the CSP headers.

To do that, we add a very simple nonce generator using Uuid. This Uuid is the same for all the request. It is used by a Twig extension to add it in templates and by a response subscriber to add it in the headers.

To sum up the request process:
- twig templates are rendered and the twig extension call the nonce generator. First call will generate it, subsequent calls will just use the cached value.
- before sending the response, the subscriber inject the CSP header. The headers call the nonce generator, again using the cached value
- the response is sent with the same nonce in twig templates and in the headers

### CODE REVIEW

File: src/Akeneo/Platform/config/pim.yml [hunk 1]
```diff
@@ -29,6 +29,8 @@ imports:
 parameters:
     pim_job_product_batch_size: 100
     env(AKENEO_PIM_URL): ''
+    env(UNIQUE_ID): '2726c7f26c'
```
Comment (jmleroux) [inline src/Akeneo/Platform/config/pim.yml]: Default value for the UNIQUE_ID env var in case it is not set.

File: src/Akeneo/Platform/Bundle/UIBundle/EventListener/AddContentSecurityPolicyListener.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::RESPONSE => 'addCspHeaders',
+        ];
+    }
+
+    public function addCspHeaders(FilterResponseEvent $event): void
+    {
+        $policy = "frame-src *.youtube.com";
```
Comment (jmleroux) [inline src/Akeneo/Platform/Bundle/UIBundle/EventListener/AddContentSecurityPolicyListener.php]: This whitelist should be discussed about what we can accept

**Comment by Fatma3011** [comment]
Hi, i have this problem, i can't solve it, any idea please ?


image

**Comment by Fatma3011** [comment]
i coud'nt handle this error, any solution ?

image
